lib/internal/util.js

```diff
@@ -1,18 +1,65 @@
 'use strict';
 
-exports.printDeprecationMessage = function(msg, warned) {
+const util = require('util');
+const warnedLocations = new Set();
+
+function getActualCallerLocation() {
+  // If external, we split five times, because stack trace will be like this
+  //   1. Error:
+  //   2. getActualCallerLocation
+  //   3. printDeprecationMessage
+  //   4. `deprecated` function in `util.deprecate`
+  //   5. The file which actually called `util.deprecate`
+  // So, we pick the fifth entry in the stack trace and parse it
+  const location = (new Error()).stack.split('\n', 5).pop();
```

Comment:
Actually you can pass any function in the stack you have access to and it will remove up to that point from the stack. For example:
```js
function a() { return b(); }
function b() { return c(); }
function c() { return d(); }
function d() {
  var o = {};
  Error.captureStackTrace(o, c);
  return o;
}

console.log(a().stack);
```
And the output:
```
Error
    at b (...)
    at a (...)
    ...
```